cmd/agent/workspace/setup_gpg.go

```diff
@@ -0,0 +1,230 @@
+package workspace
+
+import (
+	"context"
+	"encoding/base64"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/loft-sh/devpod/cmd/flags"
+	"github.com/loft-sh/log"
+	"github.com/spf13/cobra"
+)
+
+// SetupGPGCmd holds the setupGPG cmd flags
+type SetupGPGCmd struct {
+	*flags.GlobalFlags
+
+	PublicKey  string
+	OwnerTrust string
+	SocketPath string
+}
+
+// NewSetupGPGCmd creates a new command
+func NewSetupGPGCmd(flags *flags.GlobalFlags) *cobra.Command {
+	cmd := &SetupGPGCmd{
+		GlobalFlags: flags,
+	}
+	setupGPGCmd := &cobra.Command{
+		Use:   "setup-gpg",
+		Short: "setups gpg-agent forwarding in the container",
+		Args:  cobra.NoArgs,
+		RunE: func(_ *cobra.Command, _ []string) error {
+			return cmd.Run(context.Background())
+		},
+	}
+	setupGPGCmd.Flags().StringVar(&cmd.PublicKey, "publickey", "", "GPG Public keys to import in armor form")
+	setupGPGCmd.Flags().StringVar(&cmd.OwnerTrust, "ownertrust", "", "GPG Owner trust to import in armor form")
+	setupGPGCmd.Flags().StringVar(&cmd.SocketPath, "socketpath", "", "patht to the gpg socket forwarded")
+	return setupGPGCmd
+}
+
+// will forward a local gpg-agent into the remote container
+// this works by
+//
+// - stopping remote gpg-agent and removing the sockets
+// - exporting local public keys and owner trust
+// - importing those into the container
+// - ensuring the gpg-agent is stopped in the container
+// - starting a reverse-tunnel of the local unix socket to remote
+// - ensuring paths and permissions are correctly set in the remote
+func (cmd *SetupGPGCmd) Run(ctx context.Context) error {
```

Comment:
nit: feels like there's enough in here to justify a `pkg/gpg`

Reply:
Done